src/status_im/tests/contract_test/wallet_test.cljs

```diff
@@ -0,0 +1,30 @@
+(ns status-im.tests.contract-test.wallet-test
+  (:require
+   [cljs.test :refer [deftest is]]
+   [day8.re-frame.test :as rf-test]
+   legacy.status-im.events
+   [legacy.status-im.multiaccounts.logout.core :as logout]
+   legacy.status-im.subs.root
+   status-im.events
+   status-im.navigation.core
+   status-im.subs.root
+   [status-im.tests.contract-test.utils :as contract-utils]
+   [test-helpers.integration :as h]))
+
+(defn assert-accounts-contract
+  [result]
+  (is (true? (some :wallet result)))
+  (is (true? (some :chat result)))
+  (is (= 2 (count result))))
+
+(deftest check-wallet-accounts-contract
+  (h/log-headline :create-wallet-account-test)
+  (rf-test/run-test-async
+   (h/with-app-initialized
+     (h/with-account
+       (contract-utils/call-rpc-endpoint
```

Comment:
Can we go for promises instead of callbacks? It makes writing the tests more flexible. Note that we may need to use `cljs.test/async`, as documented here https://clojurescript.org/tools/testing#async-testing

For example:

```clojure
(deftest check-wallet-accounts-contract
  (h/log-headline :create-wallet-account-test)
  (rf-test/run-test-async
   (h/with-app-initialized
    (h/with-account
     (-> (contract-utils/call-rpc :accounts_getAccounts)
         (.then assert-accounts-contract)
         (.then h/logout)
         (.then #(rf-test/wait-for [::logout/logout-method])))))))
```

Once/if we have `promesa` in the repo, we'll be able to make this look nicer.

Also, once I manage to get something decent out of this issue https://github.com/status-im/status-mobile/issues/18676, we will be able to _roughly test_ like this:

```clojure
(rf-test/run-test-async
 (p/do
   (h/init-app)
   (h/init-account)
   (p/let [result (contract-utils/call-rpc :accounts_getAccounts)]
     (assert-accounts-contract result)
     (p/do (h/logout)
           (h/wait-for [::logout/logout-method])))))
```

Reply:
Yep, im exploring promises now! 👨‍💻